provider/mem/mem.go

```diff
@@ -177,8 +194,11 @@ func (a *Alerts) GetPending() provider.AlertIterator {
 
 	go func() {
 		defer close(ch)
+		a.mtx.RLock()
```

Comment:
What if we move this lock and the `alerts` variable outside the goroutine so we can use `defer` for the unlock?

Reply:
We can, but I don't see why we should. I think I would prefer the other method, and we are unable to determine the start time of the goroutine.

Reply:
Should be unaffected as the `go` statement doesn't block.

Reply:
I mean, the goroutine is scheduled by the runtime, so there will be some differences under high load.